docs/DeveloperGuide.adoc

```diff
@@ -295,35 +528,291 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 _{More to be added}_
 
 [appendix]
-== Use Cases
+== Use Cases (To Do)
+
+(For all use cases below, the *System* is the `Powerlifting Competition Competitor` and the *Actor* is the `user`, unless specified otherwise)
+
+=== Out of competition mode
+
+[discrete]
+===== Use case: UC1 - View my competition list
+
+*MSS*
+
+1.  User requests to list all competitions in the system.
+2.  System shows a list of competitions held based on the files in the folder.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+[discrete]
+==== Use case: UC2 - Add a new competition
+
+*MSS*
+
+1. User requests for a new powerlifting competition to be started.
+2. User inputs the initial competition data (e.g. name, particpants) to be inputted.
+3. System adds the new competition to the user's competition list.
+4. System shows a success message.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. Competition data is invalid.
++
+[none]
+** 2a1. System shows an error message.
++
+Use case resume at step 2.
+
+* 2b. The competition name has already been used in user's competition list.
++
+[none]
+** 2b1. System tells user that the competition is already in his competition list.
++
+Use case ends.
+
+[discrete]
+==== Use case: UC3 - Delete a competition
+
+*MSS*
+
+1. User obtains a list of competitions by listing (<<Use case: UC1 - View my competition list, *UC1*>>)
+2. User requests for a new powerlifting competition to be deleted.
+3. System adds the new competition to the user's competition list.
+4. System shows a success message.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. User gives an invalid index.
++
+[none]
+** 2a1. System shows an error message.
++
+Use case resume at step 2.
+
+
+[discrete]
+==== Use case: UC4 - Enter competition mode for a particular competition
+
+*MSS*
+
+1. User obtains a list of competitions by listing (<<Use case: UC1 - View my competition list, *UC1*>>)
+2. User requests to enter competition mode for a particular competition.
+3. System enters competition mode for a particular competition.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. User gives an invalid index.
++
+[none]
+** 2a1. System shows an error message.
++
+Use case resume at step 2.
+
+
+[discrete]
+==== Use case: UC5 - View my athlete list
+
+*MSS*
+
+1.  User requests to list all athletes in the system.
+2.  Powerlifting Competition Coordinator shows a list of all athletes in the system.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+
+[discrete]
+==== Use case: UC6 - Search for athletes
+
+*MSS*
+
+1.  User performs a search with some constraints (e.g. id, country, age, etc.).
+2.  System lists athletes satisfying all the constraints.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. System does not find any athletes satisfying all the constraints.
++
+[none]
+** 2a1. System shows an empty list.
```

Comment:
Maybe showing an error message would be clearer for the user instead of an empty list